### PR TITLE
fix(chat): select trailing summary in single-pane after tool calls

### PR DESCRIPTION
**Pull request by snakajima**
## Summary
In single-pane (non-stack) mode, the **last** assistant message was not selected after a turn that involved tool calls — the canvas stayed anchored to the first part of the response (e.g. the *"I'll create the my-portfolio collection…"* preamble) while the final summary table was never surfaced.

**Root cause.** Assistant text streams as deltas (`server/agent/stream.ts`), and native `Bash`/`Read`/`Write` tool calls route to `toolCallHistory`, never `toolResults`. So `appendToLastAssistantText` kept seeing an assistant text card as the tail of `toolResults` and merged *every* post-tool text block onto it — collapsing a `text → Write → text → Write → summary` turn into a **single** card whose selection anchors at its first line. A reloaded session splits the same stream per persisted text entry and selects the last one (`parseSessionEntries` + `resolveSelectedUuid`), which is why it looked correct only after a refresh.

**Fix.** Track `assistantTextInterrupted` on the session: set it when a tool call lands while an assistant text card is the tail, and consume it in `applyTextEvent` so the next streamed delta opens a fresh card instead of merging. Live now matches reload — each assistant text block is its own card, and the trailing summary becomes the auto-selected canvas result.

## Changes
- `src/types/session.ts` — new `assistantTextInterrupted` flag on `ActiveSession` (documented).
- `src/utils/agent/eventDispatch.ts` — set the flag on `toolCall` events.
- `src/utils/session/sessionHelpers.ts` — skip the append (open a fresh card) when interrupted; clear the flag.
- `src/utils/session/sessionFactory.ts` / `sessionEntries.ts` — initialize the flag.
- Tests: new `test/utils/session/test_assistantTextInterrupt.ts`; fixture updates in `test_skillTagging.ts` / `test_mergeSessions.ts`.

## Test plan
- [x] `yarn test` (new regression test replays the bug's `text → tool → text → tool → summary` sequence and asserts separate cards + summary selected)
- [x] `yarn lint` / `yarn typecheck` / `yarn build`
- [ ] Manual: in single-pane mode, run the Investor *"Create a my-portfolio collection…"* query and confirm the canvas lands on the final summary, not the preamble.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Ensure assistant text following tool calls is split into separate cards so the final summary message is auto-selected in single-pane mode.

Bug Fixes:
- Fix single-pane chats where post-tool assistant text was merged into a single card, preventing the trailing summary from being selected.

Enhancements:
- Track assistant-text interruption state on sessions so streaming behavior matches reloaded sessions in how assistant messages are segmented.

Tests:
- Add regression tests covering assistant text streaming across tool-call boundaries and update existing session-related tests for the new session flag.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed chat message display when tool calls interrupt assistant text streaming; subsequent text now appears in a separate message instead of merging with the previous text.
  * Ensured live-stream chat behavior matches saved session reload behavior for consistent message splitting at tool call boundaries.

[Review Change Stack](https://app.coderabbit.ai/change-stack/receptron/mulmoclaude/pull/1537?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/utils/agent/eventDispatch.ts [hunk 1]
```diff
@@ -42,6 +42,10 @@ export async function applyAgentEvent(event: SseEvent, ctx: AgentEventContext):
   switch (event.type) {
     case EVENT_TYPES.toolCall:
       session.toolCallHistory.push(toToolCallEntry(event));
+      // A tool call closes the current assistant text block: the next
+      // streamed delta must open a fresh card, not merge onto the
+      // pre-tool prose. See `ActiveSession.assistantTextInterrupted`.
+      session.assistantTextInterrupted = true;
```
[the remark this answers is not in the document]
Comment (snakajima) [reply src/utils/agent/eventDispatch.ts:48]: `toolCall` is the correct (and earliest) signal here. Claude emits the `tool_use` content block — which `stream.ts` maps to a `toolCall` event — *before* the tool runs and *before* any resumed prose. Once the flag is set, the next assistant text delta opens a fresh card regardless of whether the `toolCallResult` has arrived yet, so there's no window where post-result prose could merge onto pre-tool text. `toolCallResult` always trails its `toolCall`, so keying off it would be strictly later and redundant. Leaving it on `toolCall`.
